### ISSUE TITLE
tf.keras.layers.experimental.preprocessing.RandomXXX API is missing "nearest" as an option for fill_mode.

### ISSUE THREAD

**Issue by kechan** [tensorflow/tensorflow#39177]
**System information**
- TensorFlow version (you are using): 2.2.0-dev20200504
- Are you willing to contribute it (Yes/No): Yes (but really depends on the technicality)

**Describe the feature and the current behavior/state.**
tf.keras.layers.experimental.preprocessing.RandomTranslation's argument "fill_mode" should accept the value of "nearest". This should be available to all relevant data augmentation experimental API that will inherit from Layer. fill_mode of "nearest" is the default for tf.keras.preprocessing.image.ImageDataGenerator. 

**Will this change the current api? How?**
This is about adding an extra fill_mode option for "nearest". It should be there since this is the default for the older Keras API (for data augmentation). This will allow apple to apple comparison to reproduce results.

**Who will benefit with this feature?**

**Any Other info.**

**Comment by tanzhenyu** [comment]
you can set 'nearest' to `interpolation`

**Comment by tanzhenyu** [comment]
To be more specific:
 -- we have `fill_mode` which stands for when the output coordinate is out of range, how do we compute a in-range input coordinate, so `reflect`, `wrap`, `constant` are valid modes.
 -- we have `interpolation` which stands for when the computed equivalent input coordinate is a float, not int. In this case, `nearest` and `bilinear` are valid modes. -- potential extra mode is spline interpolation, which is what scipy supports today.

**Comment by kechan** [comment]
@tanzhenyu If I used the old keras data augmentation API for a while now, I am pretty sure I don't want interpolation to "nearest". I really want fill_mode to be "nearest". The effect is that, for example, if you right shift the image, the "void" is filled by copying the edge pixel of the photo over and over until you fill up that void. 

Keras actually used this as the default. I may guess that most ppl find it reasonable choice for most image tasks. So leaving this out of the new API is not ideal.

**Comment by tanzhenyu** [comment]
> @tanzhenyu I used the old keras data augmentation API for a while now, I am pretty sure I don't want interpolation to "nearest". I really want fill_mode to be "nearest". The effect is that, for example, if you right shift the image, the "void" is filled by copying the edge pixel of the photo over and over until you fill up that void.
> 
> Keras actually used this as the default. I may guess that most ppl find it reasonable choice for most image tasks. So leaving this out of the new API is not ideal.

In that case you may end up with the `main` content of your image being masked out. `reflect` in this case would honestly work better.

**Comment by kechan** [comment]
@tanzhenyu Not sure what you mean by "masked out" exactly. But minor masking out is sometimes what data augmentation intends to do??, e.g. crop out, or randomly obscuring part of object. Also, optimal image augmentation is domain specific, so don't need to say which work better or worse. FYI: I did try "reflect" for my task, and i obtained slightly worse accuracy (although I can't robustly attribute it to this choice. 

This feature is after all in Keras for over 2 years and it is the default. If the broader TF community thing it ought to not exist, I am fine with that. I will just have to implement my own...

**Comment by tanzhenyu** [comment]
> @tanzhenyu Not sure what you mean by "masked out" exactly. But minor masking out is sometimes what data augmentation intends to do??, e.g. crop out, or randomly obscuring part of object. Also, optimal image augmentation is domain specific, so don't need to say which work better or worse. FYI: I did try "reflect" for my task, and i obtained slightly worse accuracy (although I can't robustly attribute it to this choice.
> 
> This feature is after all in Keras for over 2 years and it is the default. If the broader TF community think it ought to not exist, I am fine with that. I will just have to implement my own...

What do you mean by default for Keras?

**Comment by kechan** [comment]
@tanzhenyu 

https://keras.io/api/preprocessing/image/#imagedatagenerator-class

**Comment by tanzhenyu** [comment]
> @tanzhenyu
> 
> https://keras.io/api/preprocessing/image/#imagedatagenerator-class
> 
> tf.keras.preprocessing.image.ImageDataGenerator(
> ...
> channel_shift_range=0.0,
> fill_mode="nearest",
> cval=0.0,
> ...
> )

Makes sense. We could support this mode, but it wouldn't be default, and probably I wouldn't be able to make it before 2.3. If you need this, contribution welcome!

**Comment by kechan** [comment]
@tanzhenyu Thanks. I am not sure when I could spend time on this due to other things I need to do. I actually tried used "constant" in my current task and it is performing about the same as "nearest", so it isn't a big issue. 

If I come around to try implement this, I will post a PR.

**Comment by tanzhenyu** [comment]
This is fixed by @WindQAQ
